e2e/e2e_test.go

```diff
@@ -81,23 +97,253 @@ var _ = Describe("Whereabouts functionality", func() {
 			})
 
 			It("allocates a single pod within the correct IP range", func() {
+				const ipv4TestRange = "10.10.0.0/16"
+
 				By("checking pod IP is within whereabouts IPAM range")
 				secondaryIfaceIP, err := secondaryIfaceIPValue(pod)
 				Expect(err).NotTo(HaveOccurred())
 				Expect(inRange(ipv4TestRange, secondaryIfaceIP)).To(Succeed())
 			})
 		})
+
+		Context("Replicaset tests", func() {
+
+			var (
+				replicaCount int32
+				k8sClient    *kubeClient.Client
+				label        string
+				k8sIPAM      *kubeClient.KubernetesIPAM // - may be used for GetIPPools - todo
+			)
+
+			BeforeEach(func() {
+				By("creating a replicaset with whereabouts net-attach-def")
+
+				replicaCount = int32(0) // begin test with 0 replicas.
```

Comment:
I would go as far as to expose a constant in the `"Replicaset tests"` context:
- emptyReplicaSet / noReplicas = 0

and use that instead. No need for the `replicaCount` variable at all.